src/sapling/saplingscriptpubkeyman.cpp

```diff
@@ -53,23 +53,24 @@ void SaplingScriptPubKeyMan::UpdateSaplingNullifierNoteMapWithTx(CWalletTx& wtx)
         SaplingOutPoint op = item.first;
         SaplingNoteData nd = item.second;
 
-        if (nd.witnesses.empty()) {
+        if (nd.witnesses.empty() || !nd.IsMyNote()) {
```

Comment:
not following the rationale here, `mapSaplingNullifiersToNotes` is never loaded with the outgoing note. Wouldn't make more sense to just continue the loop if the note isn't an incoming note?

Reply:
No.. because we don't update the nullifiers for "external" notes  (because we don't have them).
In fact... when you get to the `} else {`  branch, as you can see, you need the note's `ivk` (which you don't  have for "external" notes).